### PR TITLE
Initial version of command line program to scrape emails

### PR DESCRIPTION
**Pull request by jasmeet107**

### CODE REVIEW

File: find_email_addresses.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      mailto_content = email_link.split(":", 1)[1]
+      email_address = mailto_content.split("?", 1)[0]
+      if email_address not in email_addresses_found:
+        print email_address
+      email_addresses_found.append(email_address)
+    finished_pages.append(str(page_url))
+
+
+domain_name = sys.argv[1]
+find_emails(domain_name)
```
Comment (julien-jana) [inline find_email_addresses.py]: It's a good practice to use a main() method and argparse here. Can you rewrite it? Also, can you think of any other arguments we would be interested in?

File: find_email_addresses.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    starting_page = parsed.geturl()
+    domain_name = parsed.netloc
+  else: 
+    starting_page = 'http://' + website
+    parsed_page = urlsplit(starting_page)
+    domain_name = parsed_page.netloc
+  #Create a queue of pages that need to be checked, 
+  #and a list of pages that have already been checked
+  all_pages = [starting_page]
+  finished_pages = []
```
Comment (julien-jana) [inline find_email_addresses.py]: Can you think of better data structures than lists here?

File: find_email_addresses.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  #Keep track of email addresses that have been printed already
+  email_addresses_found = []
```
Comment (julien-jana) [inline find_email_addresses.py]: Same here, should this be a list?

File: find_email_addresses.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    try: 
+      #Make sure the page is another webpage and not a file
+      page_headers = requests.head(page_url).headers
+      if page_headers.get('content-type'):
+        header = page_headers.get('content-type').lower()
+        if not header.startswith('text/html') and not header.startswith('application/xhtml+xml'):
+          finished_pages.append(str(page_url))
+          continue
+      page = requests.get(page_url)
+    except (exceptions.SSLError, exceptions.TooManyRedirects, exceptions.ConnectionError, exceptions.InvalidURL):
```
Comment (julien-jana) [inline find_email_addresses.py:53]: Well done handling specific exceptions

File: find_email_addresses.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+  email_addresses_found = []
+  while len(all_pages) > 0: 
+    page_url = all_pages.pop(0)
+    #If we've already checked the page, skip it
+    if page_url in finished_pages:
+      continue
+    #print page_url
+    try: 
+      #Make sure the page is another webpage and not a file
+      page_headers = requests.head(page_url).headers
```
Comment (julien-jana) [inline find_email_addresses.py:46]: Neat

**Review by julien-jana — COMMENTED** [review]
Nice start. The program takes quite a while to run on web.mit.edu. Can you think of a way to limit how long it runs?

File: find_email_addresses.py (hunk 3 above)
Comment (jasmeet107) [reply find_email_addresses.py]: For both of those, are you referring to a set, or should I be thinking a little more?

**Comment by jasmeet107** [comment]
If we limit the amount of time it takes to run, that means we'll only be getting the email addresses from pages discovered during that time. Maybe a good additional argument would be a time limit.

File: find_email_addresses.py (hunk 1 above)
Comment (jasmeet107) [reply find_email_addresses.py]: Will do! Besides the time limit argument, it might also be good to have an argument for the maximum number of emails.

**Pull request closed without merging**